src/main/java/org/assertj/core/api/SoftAssertion.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import java.util.function.Consumer;
+
+import org.opentest4j.MultipleFailuresError;
+
+/**
+ * Parent interface for soft assertion implementations.
+ * 
+ * @author Fr Jeremy Krieg
+ * @see AbstractSoftAssertions
+ * @see SoftAssertions
+ */
+public interface SoftAssertion extends ThrowableCollector {
+
+  public interface ThrowingRunnable {
+    void run() throws Exception;
+  }
+
+  /**
+   * Creates a proxied assertion class of the given type. The returned value
+   * is an assertion object compatible with the supplied assertion class, but
+   * instead of throwing errors it will collect them and store them using
+   * {@link #collectError(Throwable) collectError()}.
+   * 
+   * @param <V> The type of the assertion class
+   * @param <T> The type of the object-under-test
+   * @param assertClass Class instance for the assertion type.
+   * @param actualClass Class instance for the type of the object-under-test.
+   * @param actual The actual object-under-test. 
+   * @return A proxied assertion class for the given object-under-test.
+   */
+  <V extends Assert<? extends V, ? extends T>, T> V proxy(Class<V> assertClass, Class<T> actualClass, T actual);
```

Comment:
use `ASSERT` instead of `V` and `ACTUAL` instead of `T`

Reply:
Done. Because this was a problem that I inherited I separated this change into a separate commit.